be/src/cloud/cloud_full_compaction.cpp

```diff
@@ -52,13 +52,26 @@ CloudFullCompaction::~CloudFullCompaction() = default;
 
 Status CloudFullCompaction::prepare_compact() {
     if (_tablet->tablet_state() != TABLET_RUNNING) {
-        return Status::InternalError("invalid tablet state. tablet_id={}", _tablet->tablet_id());
+        Status res = Status::InternalError("invalid tablet state. tablet_id={}", _tablet->tablet_id());
```

Comment:
Use defer samentic to avoid duplicated code, e.g.
```
Defer  defer_set_st([&failed, &st] {
    if (!failed) return;
    cloud_tablet()->set_last_full_compaction_failure_time(UnixMillis());
    cloud_tablet()->set_last_full_compaction_status(st.to_string());
})
```

Reply:
done